pkg/streamingpromql/operators/function_over_range_vector.go

```diff
@@ -79,46 +73,139 @@ func (m *FunctionOverRangeVector) NextSeries(ctx context.Context) (types.Instant
 			return types.InstantVectorSeriesData{}, err
 		}
 
-		head, tail := m.floatBuffer.UnsafePoints(step.RangeEnd)
```

Comment:
This method is getting quite long - perhaps splitting out everything from here down to line 173 into its own `computeNextStep` method would help make this more manageable? (it also sets things up nicely for supporting functions other than `rate`)